src/modules/file/file.usecase.ts

```diff
@@ -155,32 +159,68 @@ export class FileUseCases {
     );
   }
 
+  @UsePipes(
+    new ValidationPipe({
+      whitelist: true,
+    }),
+  )
   async updateFileMetaData(
     user: User,
     fileUuid: File['uuid'],
-    newFileMetada: UpdateFileMetaDto,
+    newFileMetadata: UpdateFileMetaDto,
   ) {
+    if (!newFileMetadata || Object.keys(newFileMetadata).length === 0) {
+      throw new BadRequestException('Missing update file metadata');
+    }
+    if (
+      isStringEmpty(newFileMetadata.plainName) &&
+      isStringEmpty(newFileMetadata.type)
+    ) {
+      throw new BadRequestException('Filename cannot be empty');
+    }
+
     const file = await this.fileRepository.findOneBy({
       uuid: fileUuid,
       status: FileStatus.EXISTS,
+      userId: user.id,
     });
+    if (!file) {
+      throw new NotFoundException('File not found');
+    }
 
-    if (!file.isOwnedBy(user)) {
-      throw new ForbiddenException('This file is not yours');
+    let plainName =
```

Comment:
For instance, `isOwnedBy` is a method thought like that. Here, you are increasing the code complexity of a context that should be centralised in the file entity. If you have any other situation where this method cannot be reused, you will need copy-paste all these checks.

Reply:
In this case I didnt add checks for the owner, I just removed that isOwnedBy check because I added the userId to the file.findOneBy, trying to speed the SQL petition by adding the owner filter directly to the query.
But if there is no improvement, I can revert it if it helps to the context

Reply:
Updating the metadata, like the name or the type is not an operation that could be a reason to complain about speeds, hence, one select more or less is not worth it if it is at the expense of making the code less clear or hiding the business rules or the security checks